api/src/main/java/org/openmrs/module/idgen/service/BaseIdentifierSourceService.java

```diff
@@ -442,4 +442,14 @@ public Long getSequenceValue(SequentialIdentifierGenerator seq) {
         return dao.getSequenceValue(seq);
     }
 
+    /**
+     * @see IdentifierSourceService#retireIdentifierSource(org.openmrs.module.idgen.IdentifierSource, String)
+     */
+	@Override
+	public void retireIdentifierSource(IdentifierSource identifierSource, String reason) throws APIException {
+		identifierSource.setRetired(true);
```

Comment:
Can you add a test for this functionality in the IdentifierSourceServiceTest class?
Here is an example for such a test: https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/api/LocationServiceTest.java#L449-L480

Reply:
added